File: API/Database/DbModels/Rider.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;


namespace API.Database.DbModels;

public class Rider
{
    [Key]
    public int Id { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string RiderNumber { get; set; }
    public string Category { get; set; }
    public string DateOfBirth { get; set; }
    [JsonIgnore]
    public List<Result> Results { get; set; }
    public List<Competition> Competitions { get; set; }
    public List<Image> Images { get; set; }
}

File: client/src/app/modules/administration/models/Rider.ts
import { Horse } from "./Horse";

export class Rider {
    public static clone(r: Rider): Rider {
        return new Rider(r.id, r.riderNumber, r.firstName, r.lastName, r.dateOfbirth, r.category, r.horses.map(h => Horse.clone(h)))
    }

    constructor(
        public id: number = 0,
        public riderNumber: string = "",
        public firstName: string = "",
        public lastName: string = "",
        public dateOfbirth: string = "",
        public category: string = "",
        public horses: Horse[] = []
    ) {
    }
}